src/common/session/session-entity.js

```diff
@@ -0,0 +1,223 @@
+      // if "inactive" timer is expired at "read" time -- esp. initial read -- reset
+      if (this.isExpired(obj.inactiveAt)) return this.reset()
+
+      return obj
+    } catch (e) {
+      warn('Failed to read from storage API', e)
+      // storage is inaccessible
+      return {}
+    }
+  }
+
+  /**
+   * Store data to the storage API tied to this entity
+   * To preseve existing attributes, the output of ...session.read()
+   * should be appended to the data argument
+   * @param {Object} data
+   * @returns {Object}
+   */
+  write (data) {
+    try {
+      if (!data || typeof data !== 'object') return
+      Object.keys(data).forEach(k => {
+        this[k] = data[k]
+      })
+      this.storage.set(this.lookupKey, stringify(this.compress(data)))
+      return data
+    } catch (e) {
+      // storage is inaccessible
+      warn('Failed to write to the storage API', e)
+      return null
+    }
+  }
+
+  reset () {
+    // this method should set off a chain of actions across the features by emitting 'new-session'
+    // * send off pending payloads
+    // * stop recording (stn and sr)...
+    // * delete the session and start over
+    try {
+      if (this.initialized) ee.get(this.agentIdentifier).emit('session-reset')
+      this.storage.remove(this.lookupKey)
+      this.abortController?.abort()
+      this.inactiveTimer?.clear?.()
+      this.expiresTimer?.clear?.()
+      delete this.custom
+      delete this.value
+      const newSess = new SessionEntity({
+        agentIdentifier: this.agentIdentifier,
+        key: this.key,
+        storageAPI: this.storage,
+        expiresMs: this.expiresMs,
+        inactiveMs: this.inactiveMs
+        // value: value === '0' ? value : undefined // add this back in if we have to send '0' for disabled cookies
+      })
+      Object.assign(this, newSess)
+      return newSess.read()
+    } catch (e) {
+      return {}
+    }
+  }
+
+  /**
+   * Refresh the inactivity timer data
+   */
+  refresh () {
+    if (!this.expiresTimer.isValid()) this.reset()
+    this.inactiveAt = this.getFutureTimestamp(this.inactiveMs)
+    this.write({ ...this.read(), inactiveAt: this.inactiveAt })
+  }
+
+  /**
+   * @param {number} timestamp
+   * @returns {boolean}
+   */
+  isExpired (timestamp) {
+    return Date.now() > timestamp
+  }
+
+  /**
+   * @param {Object} data
+   * @returns {boolean}
+   */
+  isInvalid (data) {
+    const requiredKeys = ['value', 'expiresAt', 'inactiveAt']
+    return !requiredKeys.every(x => Object.keys(data).includes(x))
+  }
+
+  /**
+   * @param {number} futureMs - The number of ms to use to generate a future timestamp
+   * @returns {number}
+   */
+  getFutureTimestamp (futureMs) {
+    return Date.now() + futureMs
+  }
+
+  compress (obj) {
+    // no compression applied at this time
+    // could use bel-serializer encoding here if prioritized
+    return obj
+  }
+
+  decompress (obj) {
+    // no need to decompress if we aren't compressing
+    return obj
+  }
```

Comment:
Can we remove this code since it's not being used? Seems a waste of bytes and stack.